assignment3/wordSearch.py

```diff
@@ -0,0 +1,87 @@
+from dictionary import Dictionary
+from collections import deque
+
+
+class GridWord:
+    """ A class that represents a path in the grid and a corresponding word.
+
+    Fields:
+        _last_position:    a tuple of length 2 that represents the coordinates
+                           of the last position in the path;
+        _visited_position: a set of tuples that contains all coordinates
+                           in this path;
+        _word:             a string, the word that corresponds to the path.
+    """
+    def __init__(self, last = None, visited = None, word = ''):
+        self._last_position = last
+        if visited is None:
+            self._visited_positions = set()
+        else:
+            self._visited_positions = visited
+        self._word = word
+
+    def getLastPosition(self):
+        return self._last_position
+
+    def getWord(self):
+        return self._word
+
+    def newPosition(self, i, j, symbol):
+        self._last_position = (i, j)
+        self._visited_positions.add(self._last_position)
+        self._word += symbol
+
+    def containsPosition(self, i, j):
+        return (i, j) in self._visited_positions
+
+    def copy(self):
+        return GridWord(self._last_position,
+                        self._visited_positions.copy(), self._word)
+
+
+def wordSearch(rows, cols, grid, dictionary):
+    """Given a grid of letters and a dictionary, find all the words from the
+    dictionary that can be formed in the grid. Not case-sensitive.
+
+    Args:
+        rows: an int, number of rows in the grid;
```

Comment:
Right, but `rows` and `cols` was in the assignment description: "Your function receives the number of rows, number of columns, a 2-dimensional array of
characters (of the native char data type), and the dictionary.".
For the future, we can igonre such input data, right?